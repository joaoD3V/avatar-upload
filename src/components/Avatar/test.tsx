import 'jest-canvas-mock';

import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import { Avatar } from '.';
import React from 'react';

describe('<Avatar />', () => {
  it('should render the avatar correctly', () => {
    const props = {
      src: 'https://99prod.s3.amazonaws.com/uploads/image/file/2429377/3eef648f6ce677d5e317509045df2be0.jpg',
      sliderValue: 0,
    };

    const { container } = renderWithTheme(<Avatar {...props} />);

    expect(screen.getByLabelText(/Avatar canva/i)).toBeInTheDocument();

    expect(screen.getByLabelText(/Avatar canva/i)).toHaveAttribute(
      'width',
      '113'
    );

    expect(screen.getByLabelText(/Avatar canva/i)).toHaveAttribute(
      'height',
      '113'
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
