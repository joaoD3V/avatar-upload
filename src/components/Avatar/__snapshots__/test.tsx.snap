// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render the avatar correctly 1`] = `
<canvas
  aria-label="Avatar canva"
  height="113"
  style="width: 113px; height: 113px; cursor: grab;"
  width="113"
/>
`;
